src/content/hardhat3-alpha/learn-more/configuration-variables.md

```diff
@@ -0,0 +1,288 @@
+
+## Managing encrypted variables
+
+Besides `keystore set`, Hardhat provides several other tasks to help you manage your encrypted configuration variables.
+
+- To view the value of a stored variable, use the `keystore get` task:
+
+  ::::tabsgroup{options=npm,pnpm}
+
+  :::tab{value=npm}
+
+  ```bash
+  npx hardhat keystore get SEPOLIA_RPC_URL
+  ```
+
+  :::
+
+  :::tab{value=pnpm}
+
+  ```bash
+  pnpm hardhat keystore get SEPOLIA_RPC_URL
+  ```
+
+  :::
+
+  ::::
+
+- To delete a configuration variable from the keystore, use `keystore delete`:
+
+  ::::tabsgroup{options=npm,pnpm}
+
+  :::tab{value=npm}
+
+  ```bash
+  npx hardhat keystore delete SEPOLIA_RPC_URL
+  ```
+
+  :::
+
+  :::tab{value=pnpm}
+
+  ```bash
+  pnpm hardhat keystore delete SEPOLIA_RPC_URL
+  ```
+
+  :::
+
+  ::::
+
+- To list all stored variable names, use `keystore list`:
+
+  ::::tabsgroup{options=npm,pnpm}
+
+  :::tab{value=npm}
+
+  ```bash
+  npx hardhat keystore list
+  ```
+
+  :::
+
+  :::tab{value=pnpm}
+
+  ```bash
+  pnpm hardhat keystore list
+  ```
+
+  :::
+
+  ::::
+
+## Combining environment and encrypted variables
+
+The `hardhat-keystore` plugin extends how configuration variables are resolved, but it doesn’t replace their default behavior. Hardhat still looks for environment variables first, and only falls back to the keystore if the variable isn’t set in the environment. This makes it easy to mix and override values depending on your workflow.
+
+One common use case is **overriding a keystore value with an environment variable**. For example, you might store a default value in the keystore but temporarily override it by setting an environment variable in your shell:
+
+::::tabsgroup{options=npm,pnpm}
+
+:::tab{value=npm}
+
+```bash
+SEPOLIA_RPC_URL=https://temporary-url.example npx hardhat test
+```
+
+:::
+
+:::tab{value=pnpm}
+
+```bash
+SEPOLIA_RPC_URL=https://temporary-url.example pnpm hardhat test
+```
+
+:::
+
+::::
+
+This lets you test with a different provider without changing or deleting the encrypted value.
+
+Another use case is **defining a variable exclusively through environment variables**, without using the keystore at all. This is useful in environments like CI runners. Since environment variables are always checked first, you don’t need to change anything in your config for this to work.
+
+## Under the hood
+
+The location of the encrypted keystore file depends on your operating system. You can see where it’s stored by running any keystore task with the `--verbose` flag. For example:
+
+::::tabsgroup{options=npm,pnpm}
+
+:::tab{value=npm}
+
+```bash
+npx hardhat keystore list --verbose
+```
+
+:::
+
+:::tab{value=pnpm}
+
+```bash
+pnpm hardhat keystore list --verbose
+```
+
+:::
+
+::::
+
+On Linux, this might print something like:
+
+```
+path to keystore file: /home/username/.config/hardhat-nodejs/keystore.json
+```
+
+<!--TODO: cryptography details -->
+
+<!-- TODO: how safe it is? -->
```

Comment:
@alcuadrado I think you'll have to fill in these two, I'm not familiar enough with this aspect of the keystore.

Reply:
I think we should document it the repo (i.e. `DESIGN.md`) and link that. It's super advanced for a guide.